ADAL/src/ADAuthenticationResult+Internal.m

```diff
@@ -194,8 +236,24 @@ + (ADAuthenticationResult*)resultForBrokerErrorResponse:(NSDictionary*)response
                                           errorDetails:errorDetails
                                          correlationId:correlationId];
     }
-    
-    return [ADAuthenticationResult resultFromError:error correlationId:correlationId];
+
```

Comment:
As discussed in the doc:
`In case user hasn’t updated his broker, which doesn’t have true mam feature, we need a way to tell developer that an old broker is used. Therefore, we should modify ADAL to store the broker version into the error object (i.e. userInfo of an error object) as well.`
We may need to store the broker version into the error object :)